### PR TITLE
Fix Priorities for get_active_driver

### PR DESCRIPTION
**Pull request by Emantor**
`get_active_driver` is just a special case of `get_driver`, refactor the code to
reflect that by adding an active keyword parameter. Also assert if activate and
active are both set.

This also fixes a bug where `get_active_driver` did not respect driver
priorities.

Signed-off-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>

### CODE REVIEW

File: labgrid/target.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def get_driver(self, cls, *, name=None, activate=True):
         name -- optional name to use as a filter
         activate -- activate the driver (default True)
         """
+        assert activate == active
```
Comment (jluebbe) [inline labgrid/target.py]: This breaks the default behavior (search for drivers in any state and activate them).

File: labgrid/target.py [hunk 2]
```diff
@@ -200,35 +204,7 @@ def get_active_driver(self, cls, *, name=None):
...
-            else:
-                raise NoDriverFoundError(
-                    "no active driver matching {} found in target {}".format(cls, self)
-                )
-        elif len(found) > 1:
-            raise NoDriverFoundError(
-                "multiple active drivers matching {} found in target {}".format(cls, self)
-            )
-        return found[0]
+        return self.get_driver(cls, name=name, activate=False, active=True)
```
Comment (jluebbe) [inline labgrid/target.py]: This would fail the assert above.

File: tests/test_target.py [hunk 3]
```diff
@@ -58,13 +58,13 @@ class B(Driver):
     assert "have other names" in excinfo.value.msg
     with pytest.raises(NoDriverFoundError) as excinfo:
         target[B, "adriver"]
-    assert "no active driver" in excinfo.value.msg
+    assert "no driver" in excinfo.value.msg
```
Comment (jluebbe) [inline tests/test_target.py]: I'd prefer to keep the explicit exception messages for clarity (below as well).

File: labgrid/target.py (hunk 1 above)
Comment (kjeldflarup) [reply labgrid/target.py]: Yes I encountered that!

**Comment by kjeldflarup** [comment]
Now it does not make any errors, but which driver do I then get?

**Comment by Emantor** [comment]
You get the driver with the highest priority. I'll refactor as suggested by @jluebbe.

File: labgrid/target.py (hunk 1 above)
Comment (Emantor) [reply labgrid/target.py]: Yes, that should be !=...

File: tests/test_target.py [hunk 4]
```diff
@@ -64,7 +64,7 @@ class B(Driver):
     target.activate(a2)
     with pytest.raises(NoDriverFoundError) as excinfo:
         target[A]
-    assert "multiple active drivers" in excinfo.value.msg
+    assert "multiple drivers" in excinfo.value.msg
```
Comment (jluebbe) [inline tests/test_target.py]: Why is this change still needed?

Comment (Emantor) [reply tests/test_target.py]: I missed the multiple drivers line, good catch.
